### ISSUE TITLE
Regression in partition by image range in Legion master

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#529]
The following example runs on legion master commit `cb9682e99fc553` but no longer runs on the latest master commit `b93d076c9e0b`. 

```
mport "regent"

local c = regentlib.c

fspace Cluster {
  cluster: int3d,
  bounds: rect2d,
  sep: int2d,
}

task partition_by_image_range(mat:region(ispace(int2d), double),
                              clusters:region(ispace(int3d), Cluster),
                              cluster_part:partition(disjoint, clusters, ispace(int3d)))
where
  reads(clusters.bounds)
do
  var fid = __fields(clusters)[0]

  var ip = c.legion_index_partition_create_by_image_range(__runtime(),
                                                          __context(),
                                                          __raw(mat.ispace),
                                                          __raw(cluster_part),
                                                          __raw(clusters),
                                                          fid,
                                                          __raw(clusters.ispace),
                                                          c.DISJOINT_KIND,
                                                          -1)

  var raw_part = c.legion_logical_partition_create(__runtime(), __context(), __raw(mat), ip)

  return __import_partition(disjoint, mat, clusters.ispace, raw_part)
end

task main()
  var mat = region(ispace(int2d, {400, 400}), double)

  var blocks = region(ispace(int3d, {31, 31, 64}, {1, 1, 0}), int1d)

  blocks[{25, 25, 0}] = 1
  blocks[{25, 25, 1}] = 1

  blocks[{31, 18, 0}] = 1
  blocks[{31, 18, 1}] = 1

  var allocated_blocks_part = partition(blocks, ispace(int1d, 2))
  var allocated_blocks_ispace = allocated_blocks_part[1].ispace

  var clusters = region(allocated_blocks_ispace, Cluster)
  fill(clusters.bounds, rect2d{lo = int2d{0, 0}, hi = int2d{-1, -1}})

  for color in clusters.ispace do
    clusters[color].sep = int2d{color.x, color.y}
    clusters[color].cluster = color
  end

  var cpart = partition(clusters.cluster, allocated_blocks_ispace)

  clusters[{25, 25, 0}].bounds = rect2d{ lo = int2d{0, 0}, hi = int2d{0, 0}}
  clusters[{25, 25, 1}].bounds = rect2d{ lo = int2d{0, 1}, hi = int2d{0, 3}}

  clusters[{31, 18, 0}].bounds = rect2d{ lo = int2d{380, 293}, hi = int2d{380, 296}}
  clusters[{31, 18, 1}].bounds = rect2d{ lo = int2d{381, 293}, hi = int2d{381, 296}}

  var mat_part = partition_by_image_range(mat, clusters, cpart)

  for i in mat_part.colors do
    var r = mat_part[i]
    if r.volume ~= 0 then
      var lo = r.bounds.lo
      var hi = r.bounds.hi
      c.printf("Color: %d %d %d Lo: %d %d Hi: %d %d Vol: %d\n", i.x, i.y, i.z, lo.x, lo.y, hi.x, hi.y, r.volume)
    end
  end
end

regentlib.start(main)
```

The expected output is:
```
Color: 31 18 0 Lo: 380 293 Hi: 380 296 Vol: 4
Color: 31 18 1 Lo: 381 293 Hi: 381 296 Vol: 4
Color: 25 25 0 Lo: 0 0 Hi: 0 0 Vol: 1
Color: 25 25 1 Lo: 0 1 Hi: 0 3 Vol: 3
```

On master commit `b93d076c9e0b` I am seeing this:
```
[0 - 7f64e2de0700] {5}{runtime}: [error 39] LEGION ERROR: Invalid color space color for child -9223372036678612433 of partition 2 (from file /scratch/oldhome/seshu/legion/runtime/legion/region_tree.cc:6174)[error 39] 
For more information see:
http://legion.stanford.edu/messages/error_code.html#error_code_39
```

**Comment by lightsighter** [comment]
We changed partition colors to be 32-bits. Regent needs to make sure to convert `-1` appropriately before passing it through the C interface.

**Comment by lightsighter** [comment]
Should be fixed in fcb5d2675. Pull master and try again.

**Comment by syamajala** [comment]
It works. Thanks!
